### PR TITLE
Add a first cut of the feature implementation matrix

### PR DESCRIPTION
**Pull request by AlCutter**
Add a page reflecting the current state of the various features and sub-modules of Trillian.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ X] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: docs/Feature_Implementation_Matrix.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+|:---	            | :---:   | :---:	              |:---                                                                         |
+| Spanner  	      | GA	    | ✓                   | Google internal-only, see CloudSpanner for external use.                    |
+| CloudSpanner  	| Beta 	  |   	                | Google maintains continuous-integration environment based on CloudSpanner.  |
+| MySQL  	        | GA      | ✓           	      |                                                                             |
+| Postgres  	    | In dev. |   	                | [#1298](https://github.com/google/trillian/issues/1298)                     |
+
+##### Spanner
+This is a Google-internal implementation, and is used by all of Google's current Trillian deployments.
+
+##### CloudSpanner
```
Comment (mhutchinson) [inline docs/Feature_Implementation_Matrix.md]: Might be worth rationalizing why this is in Beta. One reasonable suggestion is to raise & link to an Issue tracking blockers for this becoming GA.

File: docs/Feature_Implementation_Matrix.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+It's been tested to tens of billions of entries and tens of log tenants.
+
+Performance largely depends on the number of CloudSpanner servers allocated,
+but write throughput of 1000+ entries/s has been observed.
+
+##### MySQL
+This implementation has been tested with MySQL 5.7.
+It's currently in production use by at least one CT log operator.
+
+Write throughput of 4-500 enties/s has been observed.
```
Comment (mhutchinson) [inline docs/Feature_Implementation_Matrix.md]:
```suggestion
Write throughput of 4-500 entries/s has been observed.
```

File: docs/Feature_Implementation_Matrix.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  * Not Implemented (NI)
+  * In Development
+  * Partial
+  * Alpha
+  * Beta
+  * General Availability (GA)
+  * Deprecated (⚠)
+
+
+## Functionality
```
Comment (mhutchinson) [inline docs/Feature_Implementation_Matrix.md]: The vertical bars in (all of) your tables don't really line up. Your call on how much time you want to spend on this - it looks fine in the rendered view.

File: docs/Feature_Implementation_Matrix.md (hunk 1 above)
Comment (AlCutter) [reply docs/Feature_Implementation_Matrix.md]: Added a link to a tracking issue.

File: docs/Feature_Implementation_Matrix.md (hunk 2 above)
Comment (AlCutter) [reply docs/Feature_Implementation_Matrix.md]: Done, ta

File: docs/Feature_Implementation_Matrix.md (hunk 3 above)
Comment (AlCutter) [reply docs/Feature_Implementation_Matrix.md]: tabs vs. spaces, the eternal battle :)

**Pull request merged**